### ISSUE TITLE
Difference between 'hyper' and 'hyper-h2' and support for Requests

### ISSUE THREAD

**Issue by iitns** [python-hyper/hyper-h2#390]
Hi,
I'm implementing HTTP/2 client to send push notification request to APNS (token based)
And I'm on researching stage now, and I read an [issue on `hyper`](https://github.com/Lukasa/hyper/issues/264), which led me here.

I'm a bit confused differences between [`hyper`](http://hyper.readthedocs.io/en/latest/index.html) and [`hyper-h2`](https://python-hyper.org/h2/en/stable/index.html)
I guess `hyper-h2` is next version of `hyper` since @Lukasa shows up in both repositories,([hyper@github](https://github.com/Lukasa/hyper), [hyper-h2@github](https://github.com/python-hyper/hyper-h2)) and I can find recent commits in `hyper-h2` repository

Am I right? If not, are they intended different purpose?
(for example, `hyper` supports I/O things like [`requests`](http://docs.python-requests.org/en/master/) whereas `hyper-h2` doesn't but supports high-level interfaces)

In [I/O part in `hyper-h2` documentation](https://python-hyper.org/h2/en/stable/basic-usage.html#i-o) it seems that `hyper-h2` doesn't do I/O things so I need to implement writing/receiving data or utilize other network packages, such as [`twisted`](https://twistedmatrix.com/trac/).

`hyper` supports [Request Integration](http://hyper.readthedocs.io/en/latest/quickstart.html#requests-integration) but `hyper-h2` doesn't seem to.
My question is: is there any way that I can integrate `hyper-h2` into `requests` for client side?

Since [Best Practices for Managing Connections by Apple](https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/CommunicatingwithAPNs.html#//apple_ref/doc/uid/TP40008194-CH11-SW8) encourages using connection pool and `requests` supports it, It would be great `hyper-h2` supports `requests` integration

Thank you in advance!

**Comment by Lukasa** [comment]
hyper-h2 is not the next version of hyper. It is instead a core part of hyper. If you do `pip install hyper` you will notice that pip installs a package called `h2`: that package is hyper-h2.

hyper-h2 is intended as a building block for HTTP/2 implementations. If you're trying to integrate with APNS then you can either use something like Twisted along with hyper-h2 to write an asynchronous APNS client, or you can use `hyper` along with something like Requests to write a synchronous one. Either way works.

TL;DR: hyper-h2 isn't replacing hyper, it's a *part* of hyper.

**Comment by iitns** [comment]
Thank you for your explanation!
